## Домашнее задание №6
***

In [42]:
import numpy as np

In [43]:
import scipy.stats as stats

### Задача 1
Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

#### Решение:
Поскольку нам известно среднее квадратическое отклонение генеральной совокупности, то для решения задачи, будем использовать $Z$-критерий.\
Границы доверительного интервала расчитаем по формуле: $\displaystyle\bar X\pm z_{\alpha/2}\cdot\frac{\sigma}{\sqrt n}$, где:  
$\bar X=M=80$-среднее значение выборки    
$z_{\alpha/2}$ табличное значение $Z$-критерия, и для $95\%$-го доверительного интервала $z_{\alpha/2}=1,96$  
$\sigma=16$-среднее квадратическое отклонение генеральной совокупности  
$n=256$-объем выборки


In [44]:
M = 80
s = 16
n = 256
z_table = stats.norm.ppf(0.975)

In [45]:
left = M - z_table * s / n**0.5
rigth = M + z_table * s / n**0.5
print(f'Доверительный интервал математического ожидания с надежностью 0,95 составляет [{left:.2f}; {rigth:.2f}]')


Доверительный интервал математического ожидания с надежностью 0,95 составляет [78.04; 81.96]


***
### Ответ:
**Доверительный интервал математического ожидания составляет [78.04; 81.96]. С вероятностью 0,95 он захватит истинное среднее значение генеральной совокупности.**
***

### Задача 2
В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:\
6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1\
Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95. 

#### Решение:
$\sigma$ неизвестна, следовательно для решения задачи будем использовать t - критерий Стюдента.\
Границы доверительного интервала рассчитаем по формуле: $\displaystyle\bar X\pm t_{\alpha/2}\cdot\frac{\sigma_{несмещ}}{\sqrt{n}}$, где  
$\sigma_{несмещ}$ - несмещённое среднее квадратическое отклонение выборки, \
$n$ - размер выборки

In [46]:
x = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n = 10

In [56]:
x_avg = np.mean(x)
s = np.std(x, ddof=1)
t_tabl = stats.t.ppf(0.975, n-1)
print(f'Среднее значение по выборке: {x_avg: .2f} \n'
     f'Среднее квадратическое отклонение выборки (несмещенное): {s: .2f} \n'
     f'Табличное значение t - критерия Стьюдента для вероятности 0,95: {t_tabl: .7f}')

Среднее значение по выборке:  6.59 
Среднее квадратическое отклонение выборки (несмещенное):  0.45 
Табличное значение t - критерия Стьюдента для вероятности 0,95:  2.2621572


In [89]:
left = x_avg - t_tabl * s / n**0.5
rigth = x_avg + t_tabl * s / n**0.5
print(f'Доверительный интервал математического ожидания с надежностью 0,95 составляет [{left:.2f}; {rigth:.2f}]')

Доверительный интервал математического ожидания с надежностью 0,95 составляет [6.29; 6.89]


***
### Ответ:
**Для среднего значения по выборке 6.59 доверительный интервал с надежностью 0,95 составляет: [6.27;  6.91]**
***

### Задача 3:
Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170\
Рост матерей 178, 165, 165, 173, 168, 155, 160, 164, 178, 175\
Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

#### Решение:
Формула для нахождения доверительного интервала для разности средних арифметических: $\displaystyle\bigtriangleup\pm t_{\alpha/2}*{S_\bigtriangleup}$, где\
$\displaystyle\bigtriangleup= \bar {X_1}-\bar{X_2}$\
Стандартная ошибка разности средних:
$\displaystyle\\S_\bigtriangleup=\sqrt{\frac{D}{n_1}+\frac{D}{n_2}}$ где 
${n_1}$ и ${n_2}$ – объемы выборок.\
Формула дисперсии общей: $\displaystyle\\D=\frac{1}{2}({D_1}+{D_2})$
где ${D_1}$ и${D_2}$ – дисперсии обеих групп\
В данном случае для критерия Стьюдента степени свободы: $df = 2*(n-1)$ 

In [87]:
d = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
m = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
n = n_1 = n_2 = len(d)
print(f'Размер выборки: {n}')

Размер выборки: 10


In [80]:
d_avg = np.mean(d)
m_avg = np.mean(m)
delta = d_avg - m_avg
print(f'Средний рост дочерей: {d_avg: .1f} \n'
      f'Средний рост матерей: {m_avg: .1f} \n'
      f'Разность средних: {delta: .1f}')

Средний рост дочерей:  166.2 
Средний рост матерей:  168.1 
Разность средних: -1.9


In [86]:
D_d = np.var(d, ddof=1)
D_m = np.var(m, ddof=1)
D = 0.5*(D_d + D_m)
print(f'Несмещенная дисперсия для выборки роста дочерей: {D_d: .3f} \n'
      f'Несмещенная дисперсия для выборки роста матерей: {D_m: .3f} \n'
      f'Дисперсия общая: {D: .3f}')

Несмещенная дисперсия для выборки роста дочерей:  91.067 
Несмещенная дисперсия для выборки роста матерей:  60.100 
Дисперсия общая:  75.583


In [92]:
SE = (2 * D/n)**0.5
print(f'Стандартная ошибка разности средних: {SE: .3f}')

Стандартная ошибка разности средних:  3.888


In [78]:
t_tabl = stats.t.ppf(0.975, 2*(n - 1))
print(f'Табличное значение t - критерия Стьюдента для вероятности 0,95 и степени свободы 18: {t_tabl: .7f}')

Табличное значение t - критерия Стьюдента для вероятности 0,95 и степени свободы 18:  2.1009220


In [94]:
left = delta - t_tabl * SE
rigth = delta + t_tabl * SE
print(f'Доверительный интервал разности среднего роста дочерей и матерей с надежностью 0,95 составляет [{left:.2f}; {rigth:.2f}]')

Доверительный интервал разности среднего роста дочерей и матерей с надежностью 0,95 составляет [-10.07; 6.27]


***
### Ответ:
**Доверительный интервал разности среднего роста дочерей и матерей с надежностью 0,95 составляет [-10.07; 6.27]. Статистически значимых различий в выборках нет. Выборки статистически равны**
***